File: backend/QRFS/QRFS/Models/CitizenResponse.cs
using System;
using System.Collections.Generic;

// Code scaffolded by EF Core assumes nullable reference types (NRTs) are not used or disabled.
// If you have enabled NRTs for your project, then un-comment the following line:
// #nullable disable

namespace QRFS.Models
{
    public partial class CitizenResponse
    {
        public string Id { get; set; }
        public string QuestionId { get; set; }
        public string OptionId { get; set; }
        public string LogId { get; set; }
        public string StationId { get; set; }

        public virtual FeedbackLog Log { get; set; }
        public virtual Options Option { get; set; }
        public virtual Questions Question { get; set; }
        public virtual PoliceStation Station { get; set; }
    }
}

File: frontend/src/types/CitizenResponse.ts
import { IFeedbackLog } from "./FeedbackLog";
import { IOptions } from "./Options";
import { IPoliceStation } from "./PoliceStation";
import { IQuestion } from "./Questions";

interface Iformtype {
  questionId: string;
  optionId: string;
}

export interface ICitizenResponse {
  id?: string;
  questionId: string;
  optionId: string;
  stationId: string;
  responseDate?: string;
  logId?: string;
  feedbackLog?: IFeedbackLog;
  options?: IOptions;
  question?: IQuestion;
  policeStation?: IPoliceStation;
}
